Source/DataSources/GpxDataSource.js

```diff
@@ -0,0 +1,1022 @@
+  billboard.scaleByDistance = new NearFarScalar(
+    BILLBOARD_NEAR_DISTANCE,
+    BILLBOARD_NEAR_RATIO,
+    BILLBOARD_FAR_DISTANCE,
+    BILLBOARD_FAR_RATIO
+  );
+  billboard.pixelOffsetScaleByDistance = new NearFarScalar(
+    BILLBOARD_NEAR_DISTANCE,
+    BILLBOARD_NEAR_RATIO,
+    BILLBOARD_FAR_DISTANCE,
+    BILLBOARD_FAR_RATIO
+  );
+  billboard.verticalOrigin = new ConstantProperty(VerticalOrigin.BOTTOM);
+  billboard.image = image;
+  return billboard;
+}
+
+function createDefaultLabel() {
+  const label = new LabelGraphics();
+  label.translucencyByDistance = new NearFarScalar(3000000, 1.0, 5000000, 0.0);
+  label.pixelOffset = new Cartesian2(17, 0);
+  label.horizontalOrigin = HorizontalOrigin.LEFT;
+  label.font = "16px sans-serif";
+  label.style = LabelStyle.FILL_AND_OUTLINE;
+  return label;
+}
+
+function createDefaultPolyline(color) {
+  const polyline = new PolylineGraphics();
+  polyline.width = 4;
+  polyline.material = new PolylineOutlineMaterialProperty();
+  polyline.material.color = defined(color) ? color : Color.RED;
+  polyline.material.outlineWidth = 2;
+  polyline.material.outlineColor = Color.BLACK;
+  return polyline;
+}
+
+// This is a list of the Optional Description Information:
+//  <cmt> GPS comment of the waypoint
+//  <desc> Descriptive description of the waypoint
+//  <src> Source of the waypoint data
+//  <type> Type (category) of waypoint
+const descriptiveInfoTypes = {
+  time: {
+    text: "Time",
+    tag: "time",
+  },
+  comment: {
+    text: "Comment",
+    tag: "cmt",
+  },
+  description: {
+    text: "Description",
+    tag: "desc",
+  },
+  source: {
+    text: "Source",
+    tag: "src",
+  },
+  number: {
+    text: "GPS track/route number",
+    tag: "number",
+  },
+  type: {
+    text: "Type",
+    tag: "type",
+  },
+};
+
+let scratchDiv;
+if (typeof document !== "undefined") {
+  scratchDiv = document.createElement("div");
+}
+function processDescription(node, entity) {
+  let i;
+
+  let text = "";
+  const infoTypeNames = Object.keys(descriptiveInfoTypes);
+  const length = infoTypeNames.length;
+  for (i = 0; i < length; i++) {
+    const infoTypeName = infoTypeNames[i];
+    const infoType = descriptiveInfoTypes[infoTypeName];
+    infoType.value = defaultValue(
+      queryStringValue(node, infoType.tag, namespaces.gpx),
+      ""
+    );
+    if (defined(infoType.value) && infoType.value !== "") {
+      text = text + "<p>" + infoType.text + ": " + infoType.value + "</p>";
+    }
+  }
+
+  if (!defined(text) || text === "") {
+    // No description
+    return;
+  }
+
+  // Turns non-explicit links into clickable links.
+  text = autolinker.link(text);
+
+  // Use a temporary div to manipulate the links
+  // so that they open in a new window.
+  scratchDiv.innerHTML = text;
+  const links = scratchDiv.querySelectorAll("a");
+
+  for (i = 0; i < links.length; i++) {
+    links[i].setAttribute("target", "_blank");
+  }
+
+  const background = Color.WHITE;
```

Comment:
To be consistant with KML, I believe the background and foreground should default to `entity.balloonStyle.bgColor` and `entity.balloonStyle.textColor`, if available.

Reply:
Yeah, the whole balloon thing is custom to KML. GPX has no such visual component so this is just us looking in the file and trying to generate a default description.

Honestly, all of this might be overkill and we don't actually need to generate an Entity description at all.  Depends on what use cases we are trying to support.

Reply:
Got it. My misunderstanding from cross referencing KmlDataSource`.